service/history/historyEngine.go

```diff
@@ -2265,39 +2265,61 @@ func (e *historyEngineImpl) DeleteExecutionFromVisibility(
 	return e.visibilityMgr.DeleteWorkflowExecution(request) // delete from db
 }
 
-type updateWorkflowAction struct {
-	noop           bool
-	createDecision bool
+func (e *historyEngineImpl) updateWorkflow(
```

Comment:
We can use `updateWorkflow` to replace existing `updateWorkflowExecutionWithAction` and `updateWorkflowExecution`, since we do want to reload if runID is empty. 

I can do a refactor in another PR once this get approved